### ISSUE TITLE
EMail: Ignore TLS error doesn't suppress selfsigned cert errors

### ISSUE THREAD

**Issue by otbutz** [louislam/uptime-kuma#2999]
### ⚠️ Please verify that this bug has NOT been raised before.

- [X] I checked and didn't find similar issue

### 🛡️ Security Policy

- [X] I agree to have read this project [Security Policy](https://github.com/louislam/uptime-kuma/security/policy)

### Description

The flag `Ignore TLS error` doesn't suppress errors generated by selfsigned certificates:

![grafik](https://user-images.githubusercontent.com/22315436/228795826-99203aec-ed71-4ea7-a3cb-6a12da31d7b6.png)

![grafik](https://user-images.githubusercontent.com/22315436/228795672-67337141-3ee9-40d6-b6bf-db1c04febc41.png)

### 👟 Reproduction steps

1. Create a new notification of type `EMail (SMTP)`
2. Chose an SMTP server which uses encryption with a selfsigned certificate
3. Select `TLS` security
4. Select `Ignore TLS error`
5. Hit `Test`

### 👀 Expected behavior

The TLS error should be ignored.

### 😓 Actual Behavior

`self signed certificate in certificate chain`

### 🐻 Uptime-Kuma Version

1.21.1

### 💻 Operating System and Arch

Ubuntu 22.04

### 🌐 Browser

Firefox

### 📝 Relevant log output

```shell
Error: self signed certificate in certificate chain
    at TLSSocket.onConnectSecure (node:_tls_wrap:1539:34)
    at TLSSocket.emit (node:events:513:28)
    at TLSSocket._finishInit (node:_tls_wrap:953:8)
    at TLSWrap.ssl.onhandshakedone (node:_tls_wrap:734:12) {
  code: 'ESOCKET',
  command: 'CONN'
}
```
```

**Comment by whysi** [comment]
I've had a similar error recently, on the same version of Uptime-Kuma (1.21.1 on Alpine)

I wanted to use the internal SMTP relay server to send email notifications.
I've configured the security as "None/STARTTLS (25/587)" since the SMTP relay accepts anonymous SMTP connections on port 25 from the IP address of Uptime-Kuma (I've tested this beforehand with other tools and it worked).
I selected "Ignore TLS Error" since I wanted to use port 25 withous TLS termination but I was getting this error ("unable to get local issuer certificate"):

![image](https://user-images.githubusercontent.com/93594835/230587370-ea939464-90a5-49d2-996b-a61de5239767.png)

The thing that fixed it for me was to remove the options to "Ignore TLS Error", this way the test worked fine and I received the message:

![image](https://user-images.githubusercontent.com/93594835/230596700-ba08d875-b460-40be-a9f8-d1a7ef061e8e.png)

It seems that with that configuration selected it switched automatically on port 587 with TLS, without it is used port 25 and None. This is not so clear in the web UI or in the docs.
I'm not very good at coding but reading the source code I think that I found confirmation of this thesis here:

https://github.com/louislam/uptime-kuma/blob/f8f19d8dc5cd668665f4a4a6d3b46782fe556383/server/notification-providers/smtp.js#L11-L18

In particular line 16, where notification.smtpIgnoreTLSError can be set with that checkbox "Ignore TLS Error" in the notification setup.

I know that this is not the same situation for the bug that @otbutz had, but It's the only similar result that I found on the github issues. If needed I can open another issue to keep track also of this behaviour, I think that changing a bit the UI to explain better how to use the security as "None" on port 25 could be useful for other users.

Thanks.
